src/handlers/notebook.ts

```diff
@@ -46,28 +48,33 @@ export class DebuggerNotebookHandler implements IDisposable {
     }
     this.isDisposed = true;
     this.cellManager.dispose();
-    this.notebookTracker.activeCellChanged.disconnect(this.onNewCell);
     Signal.clearData(this);
   }
 
   protected onNewCell(noteTracker: NotebookTracker, codeCell: CodeCell) {
-    if (this.cellManager) {
-      this.cellManager.activeCell = codeCell;
-    } else {
-      this.cellManager = new CellManager({
-        breakpointsModel: this.breakpoints,
-        activeCell: codeCell,
-        debuggerModel: this.debuggerModel,
-        debuggerService: this.debuggerService,
-        type: 'notebook'
-      });
-    }
+    // need timeout casue after dispose apear bad render problem
```

Comment:
What does "bad render problem" mean here? Maybe a quick screencast could be useful to show the issue?

Reply:
Sorry, I didn't catch gif or video for that. After close notebook and go to first open notebook cell have to render box for line numbers but without number lines

Reply:
OK I'll have a look.

It's usually easier to track what is happening when connecting on signals rather than relying on `setTimeout`.

Reply:
yeah, you have absolutely right. This is some hack resolving. I guess during close widget the signal of changing cell is faster than disposing of a old handler.